### PR TITLE
Handle listeners on attach and detach for proper context switching

### LINKED ISSUE #79: Cancel pending operations and unbind listeners on detach
## Describe the bug
All methods relying on the main container being part of the DOM will lead to a multitude of errors after calling `Viewer.detach()`.

## To Reproduce
1. Use something like the create tool and start creating a shape
2. Detach the current instance
3. The preview isn't cleared and the cursor stays the same 
4. Every `MouseMove` throws an `NS_ERROR_FAILURE`

## Expected behavior
Detaching and reattaching a diagram instance should not lead to any errors.

## Additional context
Use the EventBus to announce an impending detach, so that all modules can react and prepare for it.

### PR DESCRIPTION
**Pull request by 0x4d**
Closes #79 

## Describe your changes
The main thing I did was adding the ability remove the event listeners bound to the document by the Toolbox before the instance gets detached and to rebind them after reattach.

The second step was to activate the default tool before the detach. This way, the current tool gets disabled (`onDisable()` is called). In all tool modules that use a preview, I added events to `onDisable()` to clear the various previews.

The main effect is, than no more `NS_ERROR_FAILURE`s occur after detaching the instance.

The second effect is, that you can now arbitrarily deactivate tools, even while an operation is in progress.

### CODE REVIEW

File: src/Modeler.js [hunk 1]
```diff
@@ -11,7 +11,9 @@ import MetaShapeToolsModule from './features/meta-tools';
 
 export default class Modeler extends Viewer {
 
-    constructor (options = {}) {
+    constructor (options = { canvas: { } }) {
+        options.canvas.id = 'rnw-modeler';
```
Comment (tim-kilian) [inline src/Modeler.js:15]: hmm

File: src/core/toolbox/providers/Toolbox.js [hunk 2]
```diff
@@ -79,7 +104,7 @@ export class Toolbox {
     }
 
     onMouseUp (event) {
-        if (!this.activeTool) return;
+        if (!this.activeTool || !this.mouseDown) return;
```
Comment (tim-kilian) [inline src/core/toolbox/providers/Toolbox.js:107]: can this lead in some misbehavior ?

**Review by tim-kilian — APPROVED** [review]
Overall great changes!

File: src/Modeler.js (hunk 1 above)
Comment (0x4d) [reply src/Modeler.js:15]: Do you have second thoughts about the style or about setting a default id in itself?

File: src/core/toolbox/providers/Toolbox.js (hunk 2 above)
Comment (0x4d) [reply src/core/toolbox/providers/Toolbox.js:107]: Hmm, not sure. I haven't encountered anything yet. Can you think of a case where you would want to react to a mouse up when there was no previous mouse down registered (or it was cancelled)?
